### PR TITLE
[1421] Add support for URN/UKPRN to the v2 API

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

We are going to add functionality to publish to allow providers to set the URN/UKPRN on their records. This PR lays the ground work for it. 

### Changes proposed in this pull request

- Add UKPRN field to the `provider` table
- Add URN field to the `site` table
- Expose fields on v2 serializer
- Expose fields on v2 deserializer (in order to save)

### Guidance to review

- Pull down the code and boot up a rails console 
- Run `rails db:migrate`
- Pull down a provider and change the `ukprn` to `12345678` e.g
`p = Provider.last`
`p.ukprn = 12345678`
`p.save!`
- Reload the provider and check that the new `ukprn` has persisted, e.g 
`p.reload`

- Repeat the last two steps for `urn` on the `site` table
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by avinhurry** [comment]
[[M] Add support for URN/UKPRN to the v2 API](https://trello.com/c/E8RdV7uo/1421-m-add-support-for-urn-ukprn-to-the-v2-api)

File: db/migrate/20210409124256_add_urn_to_site.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUrnToSite < ActiveRecord::Migration[6.1]
+  def change
+    add_column :site, :urn, :string
```
Comment (darokel) [inline db/migrate/20210409124256_add_urn_to_site.rb:3]: Are duplicate values allowed? ie could a provider set the same urn for different sites?

Comment (avinhurry) [reply db/migrate/20210409124256_add_urn_to_site.rb:3]: Spoke to Conor offline. Duplicate URN’s and UKPR’s are permitted

File: spec/requests/api/v2/providers/sites_spec.rb [hunk 2]
```diff
@@ -321,6 +325,12 @@ def perform_site_update
           .to(region_code)
       end
 
+      it "updates the urn of the site" do
+        expect { subject }.to change { site1.reload.urn }
+                                .from(site1.urn)
```
Comment (darokel) [inline spec/requests/api/v2/providers/sites_spec.rb:330]: Could we sort out the indentation for this?

**Review by darokel — APPROVED** [review]
One little cosmetic comment

File: spec/requests/api/v2/providers/sites_spec.rb (hunk 2 above)
Comment (avinhurry) [reply spec/requests/api/v2/providers/sites_spec.rb]: sure

**Pull request merged**